### PR TITLE
modern-sustech-thesis:0.1.0

### PR DESCRIPTION
**Pull request by Duolei-Wang**
I am submitting
- [x] a new package
- [ ] an update for a package

Description:
A template for SUSTech bachelor thesis. I construct the framework carefully so that the users can write it easily in markdown-style. Also, I preserve the config files so that one can modifies its own template from mine.

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name that isn't the most obvious or canonical name for what the package does
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] tested my package locally on my system and it worked
- [x] `exclude`d PDFs or README images, if any, but not the LICENSE
- [x] ensured that my package is licensed such that users can use and distribute the contents of its template directory without restriction, after modifying them through normal use.

### CODE REVIEW

**Comment by laurmaedje** [comment]
Thanks for the template submission! As-is the template's name is too descriptive. We have a rule that template names must include some creative/unique component so that multiple people can create a template for the same purpose or institution without one having an unfair naming advantage. See the [submission guidelines](https://github.com/typst/packages?tab=readme-ov-file#submission-guidelines) for details. You can also take a look [at the existing templates](https://typst.app/universe/search?kind=templates) for inspiration.

**Comment by Duolei-Wang** [comment]
> Thanks for the template submission! As-is the template's name is too descriptive. We have a rule that template names must include some creative/unique component so that multiple people can create a template for the same purpose or institution without one having an unfair naming advantage. See the [submission guidelines](https://github.com/typst/packages?tab=readme-ov-file#submission-guidelines) for details. You can also take a look [at the existing templates](https://typst.app/universe/search?kind=templates) for inspiration.

Thanks for your reply! I consider the similar way of some templates in typst/packages, then I name the template as modern-sustech-thesis.

**Comment by Duolei-Wang** [comment]
Rename the template as modern-sustech-thesis.

File: packages/preview/modern-sustech-thesis/0.1.0/typst.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "modern-sustech-thesis"
+version = "0.1.0"
+entrypoint = "template/main.typ"
```
Comment (laurmaedje) [inline packages/preview/modern-sustech-thesis/0.1.0/typst.toml:4]: This is not really how the template mechanism was intended. Generally, all of the formatting code should live in the package itself and not in the `template` directory and the `template` directory contains just the files that will be cloned into the user project. There you then have a `show:` rule that applies your template.

You can take a look at the `charged-ieee` template or some of the existing thesis templates to see how it is supposed to look like.

Comment (Duolei-Wang) [reply packages/preview/modern-sustech-thesis/0.1.0/typst.toml]: Thanks, I will construct my template a new version and upload it soon.

Comment (Duolei-Wang) [reply packages/preview/modern-sustech-thesis/0.1.0/typst.toml]: I finished the new version upload, please review it at your convenience!

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**